File: 0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int count = 1, l = chars.size();
        int j = 0;

        for (int i=0; i<l; i++) {
            if (i+1 < l && chars[i] == chars[i+1]) count++;
            else {
                if (count > 1) {
                    chars[j++] = chars[i];

                    for (char n : to_string(count)) chars[j++] = n;    
                }
                else chars[j++] = chars[i];
                count = 1;
            }
                
        }

        return j;
    }
};

File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] chars) {
        int j = 0;
        int count = 1; 
        for (int i = 1; i < chars.length; i++) {
            if (chars[i] != chars[i - 1]) { 
                chars[j++] = chars[i - 1]; 
                if (count > 1) {
                    for (char c : Integer.toString(count).toCharArray()) {
                        chars[j++] = c; 
                    }
                }
                count = 1;
            } else {
                count++;
            }
        }
        chars[j++] = chars[chars.length - 1];
        if (count > 1) {
            for (char c : Integer.toString(count).toCharArray()) {
                chars[j++] = c;
            }
        }
        
        return j; 
    }
}
